### DISCUSSION TITLE
Windows 7 support

### DISCUSSION THREAD

**Discussion by GentleRock** [conda-forge/miniforge#253] [Q&A]
- [x] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:

I'm on Windows 7 - 64 bits
using Miniforge3-Windows-x86_64-4.10.3-7.exe

I uninstalled any other version of Python on my desktop.

I get this error during installation,

`Failed to initialize anaconda directory`

and if I ignore it, like it was suggested here
[Anaconda3 2.4 with python 3.5 installation error (procedure entry not found; Windows 10) - Stack Overflow](https://stackoverflow.com/questions/33625683/anaconda3-2-4-with-python-3-5-installation-error-procedure-entry-not-found-win)

 the installation becomes successful.
But then when I run python, or other executables in the miniforge3 directory, I get the message

`The application was unable to start correctly (0xc000007b). Click OK to close the application. `

My guess is that some MS redistributable are missing, but which one? My guess is VS-2019. Can you confirm?

Are there other possible reasons for this error?

Thank you,

**Comment by jakirkham** [comment]
Sounds like there is an existing installation, which is causing issues. Have you tried removing existing installs and starting from scratch?

**Comment by hmaarrfk** [comment]
I think it is also possible that you have a space in your username. I wasn't able to install with a space in mine a while ago.

**Comment by GentleRock** [comment]
Thank you. for the hint.

I write few notes here, just for the record, but unfortunately I had to give up and I'll switch to a Linux distro when using Julia with Python.

The following notes are about Anaconda, both because by analyzing `miniforge` the issue I experience appears pretty much identical (in different forms it's always the error `0xc000007b`  causing this), and because I used `PythonCall` in Julia which did let me  to better isolate the issue.
[PyCall No system-wide Python was found](https://discourse.julialang.org/t/pycall-no-system-wide-python-was-found/71605/5)

I changed username, without space, and things improved in the sense that with `Python3.8 (Anaconda3-2021.05)` things worked better for a moment and I was actually able to see a plot. Just once. With `Python3.9 (Anaconda3-2021.11` things didn't work at all for me.

I analyzed the problem with the `DependencyWalker` tool. It doesn't show any significant error for what concerns `Python 3.8`, while with `Python3.9` there's a few. Maybe they are all of them related to the `API-MS-WIN-CORE-PATH-L1-1-0.DLL`, because it's a 32-bit dll and it *does* contain code, not just resources. At the same time there's no 64 bits versions around, so I really wonder, it doesn't make much sense to internally use IPC in order to communicate with a DLL that is only having functions like PathCchCanonicalize and so on, but I don't know.
I especially wonder if really nobody experiences any error of this kind under Windows7.

Best,
M

**Comment by bramson** [comment]
Yes, so this issue persists.  I couldn't find any specific claim that Windows 7 is not supported, but I get this exact same behavior.  

Running the installer produces:
`Failed to initialize anaconda directory`
and if I ignore it, the installation becomes successful...
but then when I run python, or other executables in the miniforge3 directory, I get the message
"The program can't start because api-ms-win-core-path-l1-1-0.dll is missing from your computer.  Try reinstalling the program to fix this problem"
Of course that doesn't fix anything.  

So I updated the C++ redistributable from Microsoft, nothing changed.

So I then added that specific dll file specifically.

I remove and try to reinstall miniforge and I still get the `Failed to initialize Anaconda directories` message.
And now when I run it, I get `The application was unable to start correctly (oxc000007b). Click OK to close the application.`

So it's a complete failure and waste of 2 hours.
I'll try WinPython and/or other distributions, but it's a shame that this error persists, and without even any serious comments or suggestions to fix it.

**Reply by ghost** [reply]
> Yes, so this issue persists. I couldn't find any specific claim that Windows 7 is not supported, but I get this exact same behavior.
> 
> Running the installer produces: `Failed to initialize anaconda directory` and if I ignore it, the installation becomes successful... but then when I run python, or other executables in the miniforge3 directory, I get the message "The program can't start because api-ms-win-core-path-l1-1-0.dll is missing from your computer. Try reinstalling the program to fix this problem" Of course that doesn't fix anything.
> 
> So I updated the C++ redistributable from Microsoft, nothing changed.
> 
> So I then added that specific dll file specifically.
> 
> I remove and try to reinstall miniforge and I still get the `Failed to initialize Anaconda directories` message. And now when I run it, I get `The application was unable to start correctly (oxc000007b). Click OK to close the application.`
> 
> So it's a complete failure and waste of 2 hours. I'll try WinPython and/or other distributions, but it's a shame that this error persists, and without even any serious comments or suggestions to fix it.

Use this api-ms-win-core-path DLL which is specifically designed for python on win7. Don't download a random DLL from somewhere else, it won't work - delete whatever DLL you downloaded. If python support is the only issue, this should fix your problem. Let me know if you still have issues and I'll fix the problems for you.
https://github.com/nalexandru/api-ms-win-core-path-HACK

**Reply by szc11121** [reply]
> > Yes, so this issue persists. I couldn't find any specific claim that Windows 7 is not supported, but I get this exact same behavior.
> > Running the installer produces: `Failed to initialize anaconda directory` and if I ignore it, the installation becomes successful... but then when I run python, or other executables in the miniforge3 directory, I get the message "The program can't start because api-ms-win-core-path-l1-1-0.dll is missing from your computer. Try reinstalling the program to fix this problem" Of course that doesn't fix anything.
> > So I updated the C++ redistributable from Microsoft, nothing changed.
> > So I then added that specific dll file specifically.
> > I remove and try to reinstall miniforge and I still get the `Failed to initialize Anaconda directories` message. And now when I run it, I get `The application was unable to start correctly (oxc000007b). Click OK to close the application.`
> > So it's a complete failure and waste of 2 hours. I'll try WinPython and/or other distributions, but it's a shame that this error persists, and without even any serious comments or suggestions to fix it.
> 
> Use this api-ms-win-core-path DLL which is specifically designed for python on win7. Don't download a random DLL from somewhere else, it won't work - delete whatever DLL you downloaded. If python support is the only issue, this should fix your problem. Let me know if you still have issues and I'll fix the problems for you. https://github.com/nalexandru/api-ms-win-core-path-HACK

I copy api-ms-win-core-path-l1-1-0.dll to the C:\Windows\SysWOW64 and C:\Windows\System32 but still got "Failed to initialize conda directories" at last

**Comment by hmaarrfk** [comment]
I'm sorry, but windows 7 is quite old and I don't think anybody has plans to help support it. I will move this issue to a discussion in case you want to help document your adventure, but if you are trying to get work done, I suggest updating to windows 10 (paid option) or installing Ubuntu (free option)

**Reply by Hueristic** [reply]
> I'm sorry, but windows 7 is quite old and I don't think anybody has plans to help support it. I will move this issue to a discussion in case you want to help document your adventure, but if you are trying to get work done, I suggest updating to windows 10 (paid option) or installing Ubuntu (free option)

Your comment is less than worthless, suggesting someone change operating systems to use something that is supposed to be supported is absurd.

**Reply by hmaarrfk** [reply]
Reading my answer 2 years later, i can see how it might have been perceived as confrontational. I apologize for the tone.

In 2024, I would not be moving it to a discussion, I feel like the distinction between github discussions and issues is quite thin and discussions have not caught on for me personally.

However, please consider the fact that Windows 8, the successor of Windows 7 was released in 2012, that was more than 10 years before my original reply. Windows has gotten better at offering users "updates" to newer versions for free. conda-forge is a community of volunteers and I'm not sure "supposed to be supported" means in the context of conda-forge's packages. The omission of information can also be due to the fact that "we just aren't sure if it is supported".

I use ubuntu on a daily basis and it is really a "good alternative" for somebody who is stuck on Windows 7 trying to run python from our sources.

We have quite a few windows specific issues that many maintainers do not have much expertise in. I've been tagging them here
https://github.com/conda-forge/miniforge/issues?q=is%3Aopen+is%3Aissue+label%3Awindows

If you have constructive ways to help those users with conda-forge's software, please let us know by replying to the appropriate issue where you can contribute.

**Reply by Hueristic** [reply]
I have been fighting for 30 years against M$'s forced version changes and sometimes come across crass if I see what looks like people being pushed into their corporate monopolistic practices so thx for the honest reply and I hope you find the needed help. It seems other projects are using this as a dependency and if it doesn't stay backwards compatible then those projects by default play into m$'s hand.

**Comment by schloegl** [comment]
There requirements on this page [1] say that Windows>=7, on acrhictecture x76_64 is installed. When I tried to run the miniforge installers between v22 and v25, every attempted failed. Is anyone else able to install Miniforge3 on Win7? If not, the web page [1] should be updated and correct for the supported versions of windows.  

[1] https://github.com/conda-forge/miniforge

**Reply by hmaarrfk** [reply]
See: https://github.com/conda-forge/miniforge/discussions/253#discussioncomment-2027867

Happy to merge PRs that update the information to the "truth" as of today.
